### PR TITLE
pcre2: add more source mirrors

### LINKED ISSUE #7963: pcre2/10.37: fails to build due to source files download connection timeout
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: pcre2/10.37
  * Operating System+version: macOS 12.0.1
  * Compiler+version: clang 13.0
  * Conan version: 1.42.0
  * Python version: 3.9.7

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.0
os=Macos
os_build=Macos
[options]
[build_requires]
[env]
```

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
pcre2/10.37: Configuring sources in ~/.conan/data/pcre2/10.37/_/_/source
ERROR: Error downloading file https://ftp.pcre.org/pub/pcre/pcre2-10.37.tar.gz: 'HTTPSConnectionPool(host='ftp.pcre.org', port=443): Max retries exceeded with url: /pub/pcre/pcre2-10.37.tar.gz (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x102eadf40>, 'Connection to ftp.pcre.org timed out. (connect timeout=60.0)'))'
Waiting 5 seconds to retry...
ERROR: pcre2/10.37: Error in source() method, line 76
	tools.get(**self.conan_data["sources"][self.version],
	ConanException: Error downloading file https://ftp.pcre.org/pub/pcre/pcre2-10.37.tar.gz: 'HTTPSConnectionPool(host='ftp.pcre.org', port=443): Max retries exceeded with url: /pub/pcre/pcre2-10.37.tar.gz (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x102ead310>, 'Connection to ftp.pcre.org timed out. (connect timeout=60.0)'))'
```

</details>

### PR DESCRIPTION
**Pull request by lkotsonis**
**pcre2/10.32**
**pcre2/10.33**
**pcre2/10.35**
**pcre2/10.36**
**pcre2/10.37**

Fixes #7963 

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SpaceIm — APPROVED** [review]

**Comment by SpaceIm** [comment]
You could even download tar.bz2 instead of tar.gz to reduce archive size.

**Review by toge — APPROVED** [review]

**Review by Croydon — APPROVED** [review]

**Comment by lkotsonis** [comment]
> You could even download tar.bz2 instead of tar.gz to reduce archive size.
> 
> (`Adresses` in the not a valid keyword for github to automatically close an issue)

Updated to tar.bz2 as suggested (and used "Fixes" in description)

**Review by SpaceIm — APPROVED** [review]

File: recipes/pcre2/all/conandata.yml [hunk 1]
```diff
@@ -1,16 +1,27 @@
 sources:
   "10.32":
-    url: "https://ftp.pcre.org/pub/pcre/pcre2-10.32.tar.gz"
-    sha256: "9ca9be72e1a04f22be308323caa8c06ebd0c51efe99ee11278186cafbc4fe3af"
+    url:
+      - "https://sourceforge.net/projects/pcre/files/pcre2/10.32/pcre2-10.32.tar.bz2"
```
Comment (uilianries) [inline recipes/pcre2/all/conandata.yml:4]: It won't work if the checksums are different. If the first link fails, the second has a different sah256 which Conan won't allow.

Comment (uilianries) [reply recipes/pcre2/all/conandata.yml:4]: I see, you changed the tar file to bz2.  Nice!

**Review by uilianries — APPROVED** [review]
LGTM

**Review by SSE4 — APPROVED** [review]

**Pull request merged**